exchange/exchange.go

```diff
@@ -451,18 +465,25 @@ func recordImpMetrics(bidRequest *openrtb2.BidRequest, metricsEngine metrics.Met
 }
 
 // applyDealSupport updates targeting keys with deal prefixes if minimum deal tier exceeded
-func applyDealSupport(bidRequest *openrtb2.BidRequest, auc *auction, bidCategory map[string]string) []error {
+func applyDealSupport(bidRequest *openrtb2.BidRequest, auc *auction, bidCategory map[string]string, multiBid ExtMultiBidMap) []error {
 	errs := []error{}
 	impDealMap := getDealTiers(bidRequest)
 
 	for impID, topBidsPerImp := range auc.winningBidsByBidder {
 		impDeal := impDealMap[impID]
-		for bidder, topBidPerBidder := range topBidsPerImp {
-			if topBidPerBidder.DealPriority > 0 {
-				if validateDealTier(impDeal[bidder]) {
-					updateHbPbCatDur(topBidPerBidder, impDeal[bidder], bidCategory)
-				} else {
-					errs = append(errs, fmt.Errorf("dealTier configuration invalid for bidder '%s', imp ID '%s'", string(bidder), impID))
+		for bidder, topBidsPerBidder := range topBidsPerImp {
+			for i, topBid := range topBidsPerBidder {
+				m, ok := multiBid[bidder.String()]
+				if i > 0 && (!ok || m.TargetBidderCodePrefix == "") {
+					// apply targeting keys to only first bid if multibid is not enabled for this bidders
+					break
```

Comment:
This line isn't covered by tests, could you add a test to cover this case?

Reply:
added test cases under `TestApplyDealSupportMultiBid()`